lib/internal/process/promises.js

```diff
@@ -33,31 +25,34 @@ function setupPromises(scheduleMicrotasks) {
 
   function rejectionHandled(promise) {
     const hasBeenNotified = hasBeenNotifiedProperty.get(promise);
-    if (hasBeenNotified !== undefined) {
-      hasBeenNotifiedProperty.delete(promise);
+    hasBeenNotifiedProperty.delete(promise);
+    if (hasBeenNotified) {
```

Comment:
the `.delete` can be inside the `if` I think

Reply:
It should always be deleted, otherwise the reference would be kept. This function is only called once per handled rejection and it can not be freed otherwise.